src/modules/drawing/components/AddVertexButton.vue

```diff
@@ -0,0 +1,58 @@
+<script setup>
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import { onMounted, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { useStore } from 'vuex'
+
+import { EditMode } from '@/store/modules/drawing.store'
+// import { EditableFeatureTypes } from '@/api/features/EditableFeature.class'
+import { useTippyTooltip } from '@/utils/composables/useTippyTooltip'
+const dispatcher = { dispatcher: 'AddVertexButton.vue' }
+const i18n = useI18n()
+const store = useStore()
+
+const props = defineProps({
+    tooltipText: {
+        type: String,
+        default: 'modify_add_vertex',
+    },
+    // If true, the button will add a vertex in the reverse direction
+    reverse: {
+        type: Boolean,
+        default: false,
+    },
+})
+useTippyTooltip('#addVertexButton [data-tippy-content]', { placement: 'left' })
+const emit = defineEmits(['button-mounted'])
```

Comment:
I would order that a bit differently, having

- props
- emits
- definition of refs
- import of dependencies (store, i18n)
- use of composables